src/Sema.zig

```diff
@@ -7728,6 +7729,18 @@ fn zirElemType(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Ai
     return sema.addType(ptr_ty.childType(mod));
 }
 
+fn zirVectorElemType(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.Inst.Ref {
+    const mod = sema.mod;
+    const un_node = sema.code.instructions.items(.data)[inst].un_node;
+    const operand_src: LazySrcLoc = .{ .node_offset_un_op = un_node.src_node };
+    const vector_ty = try sema.resolveType(block, .unneeded, un_node.operand);
```

Comment:
In this case you want `resolveDestType` (otherwise we could leak a `GenericPoison` if the destination type is unknown due to an `anytype` parameter), and a source location is needed (`operand_src`).

Reply:
`resolveDestType` also needs builtin name which looks wrong inside `zirVectorElemType`

Reply:
Ah yes, that's not ideal, but the current code is definitely wrong.
One solution would be to rename this instruction again (sorry!) to be specifically for `@splat` - that's probably easiest.
However, if you prefer, here's a cleaner solution:
```zig
const vector_ty = sema.resolveType(block, src, zir_ref) catch |err| switch (err) {
    error.GenericPoison => return Type.generic_poison,
    else => |e| return e,
};
```
...actually, that's not quite right, either. You need to unwrap error unions and optionals like `resolveDestType` does. Feel free to split that function in two if it's convenient.
(I need to investigate more into what the `GenericPoison` return is actually for in this case - we may end up being able to propagate this for other such builtins too.)

Reply:
Okay, yeah, looks like the fix I describe there is actually correct - it's also what `zirFieldType` does for instance.

Reply:
No worries and thanks for guiding me. i think specializing instruction name to `@splat` makes sense